### PR TITLE
[frontend/backend] Cannot hide entity type "External references" (#6327)

### PR DESCRIPTION
**Pull request by SarahBocognano**
### Proposed changes

* Add missing external references in subtypes, available settings and hiddenTypesField

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/6327

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Archidoit** [comment]
In hidden types list display, 'External reference' should be displayed after 'malware analysis' (not after 'Report') : 
image


Done ✅

File: opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts [hunk 1]
```diff
@@ -45,3 +47,13 @@ test('Check Logo replacement', async ({ page }) => {
...
+
+test('Check Hidden Entities', async ({ page }) => {
+  const leftBarPage = new LeftBarPage(page);
+  const settingPage = new SettingsPage(page);
+  await page.goto('/');
+  await page.getByTestId('ChevronRightIcon').click();
+  await leftBarPage.clickOnMenu('Settings', 'Parameters');
+  await settingPage.getHiddenEntityTypeSelect().click();
+  await expect(settingPage.getExternalReference()).toBeVisible();
+});
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts]: I'm not a fan of this test. E2E tests have a high cost (~30sec in CI for a small test) and so their purpose is to validate a user scenario that is important we know it works. 

Here we are just validating the presence of an item in a list, if we really want to check those kind of things it should be done with unit test using `vite` but not in E2E.

Comment (lndrtrbn) [reply opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts]: Imo just remove it

**Review by lndrtrbn — CHANGES_REQUESTED** [review]

File: opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts (hunk 1 above)
Comment (SarahBocognano) [reply opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts]: Well that was a user scenario as it was a bug and noticed by a openCTI user himself. 
So I don't understand, do we or do we not want tests ? I think the team needs to be precise on when to do or not do tests.

Comment (lndrtrbn) [reply opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts]: I'll open a discussion between the team to discuss about it

File: opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { test } from '@playwright/test';
-import { expect } from '../fixtures/baseFixtures';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/tests_e2e/settings/parameters.spec.ts]: This comment should not be necessary anymore with the new eslint config

File: opencti-platform/opencti-front/src/private/components/settings/hidden_types/HiddenTypesField.tsx [hunk 3]
```diff
@@ -169,47 +169,52 @@ const HiddenTypesField: FunctionComponent<HiddenTypesFieldProps> = ({
...
-        <Box sx={{ display: 'flex', flexWrap: 'wrap', gap: 0.5 }}>
-          {selected.map((value) => (
-            <Chip key={value} label={t_i18n(`entity_${value}`)} />
-          ))}
-        </Box>
-      )}
-    >
-      {computeItems()}
-    </Field>
+    <div data-testid="hiddenEntityTypes">
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/settings/hidden_types/HiddenTypesField.tsx]: Not needed anymore ?

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**